dspace-api/src/main/java/org/dspace/content/BitstreamServiceImpl.java

```diff
@@ -247,7 +247,13 @@ public void delete(Context context, Bitstream bitstream) throws SQLException, Au
         context.addEvent(new Event(Event.DELETE, Constants.BITSTREAM, bitstream.getID(),
                 String.valueOf(bitstream.getSequenceID()), getIdentifiers(context, bitstream)));
 
-        bitstream.getBundles().clear();
+        //Remove our bitstream from all our bundles
+        final List<Bundle> bundles = bitstream.getBundles();
+        for (Bundle bundle : bundles) {
+            bundle.getBitstreams().remove(bitstream);
+        }
+        //Remove all bundles from the bitstream object, clearing the connection in 2 ways
+        bundles.clear();
```

Comment:
This removal of the Bundle link needs to be the *last* task in the `delete()` method. Removing this link means that Community/Collection Admins lose their access rights, and the `update()` call later on in the same method will always fail.